server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesHotThreadsAction.java

```diff
@@ -37,20 +37,46 @@
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 
-
 public class RestNodesHotThreadsAction extends BaseRestHandler {
 
+    static final String formatDeprecatedMessageWithoutNodeID = "%s is a deprecated endpoint. Please use [/nodes/hot_threads] instead.";
+    static final String formatDeprecatedMessageWithNodeID = "%s is a deprecated endpoint. Please use [/nodes/{nodeId}/hot_threads] instead.";
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_HOT_THREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_cluster/nodes/hot_threads]"); 
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOT_THREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_cluster/nodes/{nodeId}/hot_threads]");
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_HOTTHREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_cluster/nodes/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOTTHREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_cluster/nodes/{nodeId}/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_NODES_HOTTHREADS = String.format(formatDeprecatedMessageWithoutNodeID, "[/_nodes/hotthreads]"); 
+    static final String DEPRECATED_MESSAGE_NODES_NODEID_HOTTHREADS = String.format(formatDeprecatedMessageWithNodeID, "[/_nodes/{nodeId}/hotthreads]");
+
+    @Override
+    public List<DeprecatedRoute> deprecatedRoutes() {
+        final String DEPRECATED_PATH_CLUSTER_NODES_HOT_THREADS = "/_cluster/nodes/hot_threads";
+        final String DEPRECATED_PATH_CLUSTER_NODES_NODEID_HOT_THREADS = "/_cluster/nodes/{nodeId}/hot_threads";
+        final String DEPRECATED_PATH_CLUSTER_NODES_HOTTHREADS = "/_cluster/nodes/hotthreads";
+        final String DEPRECATED_PATH_CLUSTER_NODES_NODEID_HOTTHREADS = "/_cluster/nodes/{nodeId}/hotthreads";
+        final String DEPRECATED_PATH_NODES_HOTTHREADS = "/_nodes/hotthreads";
+        final String DEPRECATED_PATH_NODES_NODEID_HOTTHREADS = "/_nodes/{nodeId}/hotthreads";
```

Comment:
What's the reason for declaring all of these local variables?

Reply:
I originally thought that it is easier to read. But these local variables will not be used afterwards. So I removed them!